src/main/scala/com/kinja/amqp/persistence/MySqlMessageStore.scala

```diff
@@ -89,107 +131,120 @@ abstract class MySqlMessageStore extends MessageStore {
-						AS c)
+					c.*
+				FROM
+					rabbit_confirmations c
+				LEFT JOIN
+					rabbit_messages m
+						ON m.channelId = c.channelId
+						AND m.deliveryTag <= c.deliveryTag
+				WHERE
+					c.multiple = 1
+					AND
+					m.id IS NULL
 			"""
 
-		def selectMessageCreatedBefore(formattedTime: String, exchangeName: String, limit: Int) =
+		def selectLockedMessages(limit: Int) =
 			sql"""
 				SELECT *
 			 		FROM rabbit_messages
-					WHERE `exchangeName` = $exchangeName
-						AND `createdTime` < $formattedTime
+					WHERE `processedBy` = $processId
 		 		LIMIT $limit
-		 		FOR UPDATE
 			""".as[Message]
 
-		def selectConfByChannelIds(ids: List[String]) = {
-			val whereClause = s"""WHERE `channelId` IN (${ids.map("'" + _ + "'").mkString(",")})"""
-			sql"""
-				SELECT *
-			 		FROM rabbit_confirmations
-					#$whereClause
-			""".as[MessageConfirmation]
-		}
-
-		def selectConfByChannelAndDelivery(channelId: String, deliveryTag: Long) = for {
+		def selectConfByChannelIds(ids: List[String]) = for {
 			c <- MessageConfirmationTable
+			if c.channelId inSet ids
+		} yield c
+
+		def selectMessageByChannelAndDelivery(channelId: String, deliveryTag: Long) = for {
+			c <- MessageTable
 			if c.channelId === channelId && c.deliveryTag === deliveryTag
 		} yield c
 	}
 
 	override def saveMessage(msg: Message): Unit = {
 		Database.forDataSource(writeDs).withSession {
-			import Database.threadLocalSession
 			MessageTable.autoInc.insert(msg)
 		}
 	}
 
 	override def saveConfirmation(conf: MessageConfirmation): Unit = {
 		Database.forDataSource(writeDs).withSession {
-			import Database.threadLocalSession
 			MessageConfirmationTable.autoInc.insert(conf)
 		}
 	}
 
 	override def deleteMessageUponConfirm(channelId: String, deliveryTag: Long): Int =
 		Database.forDataSource(writeDs).withSession {
-			import Database.threadLocalSession
-			Queries.selectConfByChannelAndDelivery(channelId, deliveryTag).delete
+			Queries.selectMessageByChannelAndDelivery(channelId, deliveryTag).delete
 		}
 
-	override def deleteMultiConfIfNoMatchingMsg(olderThan: Long): Unit =
+	override def deleteMultiConfIfNoMatchingMsg(olderThanSeconds: Long): Unit =
 		Database.forDataSource(writeDs).withSession {
-			import Database.threadLocalSession
-			val df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
-			val formatted = df.format(new Date(olderThan))
+			val formatted = getFormattedDateForSecondsAgo(olderThanSeconds)
 			Queries.deleteMultiConfIfNoMsg(formatted).execute()
 		}
 
-	def createTransactionalStore: TransactionalMessageRepo = {
-		implicit val session = Database.forDataSource(writeDs).createSession()
-		new TransactionalMessageRepo()
+	override def loadLockedMessages(limit: Int): List[Message] = {
+		Database.forDataSource(writeDs).withSession {
+			Queries.selectLockedMessages(limit).list()
+		}
+	}
+
+
+	override def loadConfirmationByChannels(channelIds: List[String]): List[MessageConfirmation] = {
+		if (channelIds.isEmpty) List()
```

Comment:
Do you think we can omit this `isEmpty` check now that we use slick `inSet` in the query?